### ISSUE TITLE
Questions about the main.py code.

### ISSUE THREAD

**Issue by JeonHyeongJunKW** [Nanne/pytorch-NetVlad#52]
When the code is in clustering mode, a layer called L2Norm() is attached to the back of the existing vgg-16 and alexnet, and then clustering to obtain the cluster centroid. However, training mode does not seem to attach these layers. Within NetVLAD layer, the `forward()` function initially has this content. 

```
if self.normalize_input:
    x = F.normalize(x, p=2, dim=1) # across descriptor dim
```

Is this taking over the role? And what does this role mean?

**Comment by Nanne** [comment]
L2Norm is defined here: https://github.com/Nanne/pytorch-NetVlad/blob/master/main.py#L294-L295
as you can see it does the exact same L2 Normalisation as is done in the NetVlad layer.
